lib/ops/opsservice/update.go

```diff
@@ -260,23 +263,53 @@ func (s *site) createUpdateOperation(req ops.CreateSiteAppUpdateOperationRequest
 		return nil, trace.Wrap(err, "failed to create update operation")
 	}
 
-	resetSiteState := func() {
+	resetClusterState := func() {
 		if err == nil {
 			return
 		}
-		errReset := s.setSiteState(ops.SiteStateActive)
+
+		// Fail the operation and reset cluster state.
+		// It is important to complete the operation as subsequent same type operations
+		// will not be able to complete if there's an existing incomplete one
+		errReset := ops.FailOperation(op.Key(), s.service, trace.Unwrap(err).Error())
```

Comment:
I'm not removing the operation as I want to persist whatever operation logs that have been written to this point.